File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int[] prefix = new int[gain.length + 1];
        prefix[0] = 0;
        int max = 0;

        for (int x = 1; x < prefix.length; x++) {
            prefix[x] = gain[x - 1] + prefix[x - 1];
            max = Math.max(prefix[x], max);
        }

        return max;
    }
}

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        n = len(gain)
        prefix = [0] * (n + 1)
        m = 0

        for x in range(1, n + 1) :
            prefix[x] = gain[x - 1] + prefix[x - 1]
            m = max(prefix[x], m)

        return m